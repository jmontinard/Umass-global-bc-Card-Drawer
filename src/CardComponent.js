import React from 'react';
import { Card } from 'react-bootstrap';

const CardComponent = ({ card, angle }) => (
  <Card
    style={{
      width: '18rem',
      position: 'absolute',
      transform: `rotate(${angle}deg)`,
    }}
  >
    <Card.Img variant="top" src={card.image} />
    <Card.Body>
      <Card.Title>
        {card.value} of {card.suit}
      </Card.Title>
    </Card.Body>
  </Card>
);

export default CardComponent;
