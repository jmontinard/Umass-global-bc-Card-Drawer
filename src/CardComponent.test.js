import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import CardComponent from './CardComponent';

afterEach(cleanup);

it('renders without crashing', () => {
  const card = { image: 'test_image', value: 'test_value', suit: 'test_suit' };
  render(<CardComponent card={card} angle={0} />);
});

it('matches snapshot', () => {
  const card = { image: 'test_image', value: 'test_value', suit: 'test_suit' };
  const { asFragment } = render(<CardComponent card={card} angle={0} />);
  expect(asFragment()).toMatchSnapshot();
});
