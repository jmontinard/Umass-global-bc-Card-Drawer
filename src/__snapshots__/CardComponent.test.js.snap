// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
    style="width: 18rem; position: absolute; transform: rotate(0deg);"
  >
    <img
      class="card-img-top"
      src="test_image"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        test_value of test_suit
      </div>
    </div>
  </div>
</DocumentFragment>
`;
